OpenRA.Mods.Cnc/Traits/Infiltration/InfiltrateForCash.cs

```diff
@@ -43,6 +46,10 @@ class InfiltrateForCash : INotifyInfiltrated
 
 		void INotifyInfiltrated.Infiltrated(Actor self, Actor infiltrator)
 		{
+			var infiltrates = infiltrator.Info.TraitInfo<InfiltratesInfo>();
```

Comment:
Can you please instead change the interface to pass the type through the notification instead?
```csharp
	public interface INotifyInfiltrated { void Infiltrated(Actor self, Actor infiltrator, HashSet<string> types); }
```

This follows the convention from our other types, and makes it easier to support multiple `Infiltrates` traits on an actor.

Reply:
You can then change `Infiltrate.cs` to use
```csharp
			foreach (var t in target.TraitsImplementing<INotifyInfiltrated>())
				t.Infiltrated(target, self, infiltrates.Info.Types);
```